feat: Use state persistence mechanism for language and currency (#12076)
SAP/spartacus: docs/migration/4_0.md

# Technical Changes in Spartacus 4.0

## Breaking Changes Introduced in 4.0
- LanguageService, CurrencyService and BaseSiteService now validate the value passed to the method `setActive()` against the iso codes listed in the Spartacus `context` config, before setting the actual value in the ngrx store
- the initialization of the site context is scheduled a bit earlier than in before (now it's run in an observable stream instead of a Promise's callback). It's a very slight change, but might have side-effects in some custom implementations.

> 

Reviewer: ```suggestion
- the active language and currency are now persisted in the Session Storage instead of the Local Storage

```

# Technical Changes in Spartacus 4.0

## Breaking Changes Introduced in 4.0
- LanguageService, CurrencyService and BaseSiteService now validate the value passed to the method `setActive()` against the iso codes listed in the Spartacus `context` config, before setting the actual value in the ngrx store
- the initialization of the site context is scheduled a bit earlier than in before (now it's run in an observable stream instead of a Promise's callback). It's a very slight change, but might have side-effects in some custom implementations.
- the active language and currency are now persisted in the Local Storage instead of the Session Storage

## Automated Migrations for Version 4.0

> - `CurrencyService` no longer uses `WindowRef`. The currency initialization from the state was moved to `CurrencyStatePersistenceService`.
> - `LanguageService` no longer uses `WindowRef`. The language initialization from the state was moved to `LanguageStatePersistenceService`.

Reviewer: ```suggestion
- `CurrencyService` no longer uses `WindowRef`. The currency initialization from the state was moved to `CurrencyInitializer`.
- `LanguageService` no longer uses `WindowRef`. The language initialization from the state was moved to `LanguageInitializer`.
```

# Technical Changes in Spartacus 4.0

## Breaking Changes Introduced in 4.0

> 

Reviewer: ```suggestion
- LanguageService, CurrencyService and BaseSiteService now validate the value passed to the method `setActive()` against the iso codes listed in the Spartacus `context` config, before setting the actual value in the ngrx store
- the initialization of the site context is scheduled a bit earlier than in before (now it's run in an observable stream instead of a Promise's callback). It's a very slight change, but might have side-effects in some custom implementations.

```
